docs/book/provider_implementations/register_schemes.md

```diff
@@ -48,13 +48,15 @@ func main() {
                 panic(fmt.Errorf("GetConfigOrDie didn't die"))
         }
 
+        metricsAddr := flag.String("metrics-addr", ":8080", "The address the metric endpoint binds to.")
```

Comment:
Go flag variables should be defined outside the main function

Reply:
I've seen many examples not doing so. Is this a convention for cluster-api?

Reply:
In general I agree with @vincepri here (making the variable global in scope make it easier to reuse as a library). In this case I wonder if we should attempt to follow `kubebuilder` instead? https://github.com/cattlek8s/cluster-api-provider-generic/blob/master/cmd/manager/main.go#L38

Reply:
@davidewatson but the example here 
https://github.com/cattlek8s/cluster-api-provider-generic/blob/master/cmd/manager/main.go#L38 is using flag inside main function, so do you mean we need to keep it in main or move it out of main?

Reply:
I meant that defining the variable within the function is consistent with what `kubebuilder` does.